yt_dlp/extractor/cspan.py

```diff
@@ -242,3 +243,49 @@ def get_text_attr(d, attr):
                 'title': title,
                 'id': 'c' + video_id if video_type == 'clip' else video_id,
             }
+
+
+class CSpanCongressIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?c-span\.org/congress/'
+    IE_DESC = 'C-SPAN'
+    _TESTS = [{
+        'url': 'https://www.c-span.org/congress/?chamber=house&date=2017-12-13&t=1513208380',
+        'info_dict': {
+            'id': 'Congressional Chronicle - Members of Congress, Hearings and More | C-SPAN.org',
+            'title': 'Congressional Chronicle - Members of Congress, Hearings and More',
+            'description': 'md5:54c264b7a8f219937987610243305a84',
+            'ext': 'mp4'
+        }
+    }]
+
+    def _real_extract(self, url):
+        display_id = self._generic_id(url)
+        webpage = self._download_webpage(url, display_id)
+        file = open("sample.txt", "w")
+        file.write(webpage)
+        file.close()
+        video_url = re.findall(r'sources: \[\{file: \'(.*?)\'\}\],', webpage)[0]
```

Comment:
```suggestion
        video_url = self._search_regex(r'sources: \[\{file: \'(.*?)\'\}\],', webpage, 'video URL')
```
You can also just extract the jwplayer config with
```py
self._parse_json(
    self._search_regex(r'jwsetup\s*=\s*({.+});', webpage, 'player config'),
    video_id, transform_source=js_to_json),
```
 and use `_parse_jwplayer_data` to parse that (which should also extract captions), but ¯\\\_(ツ)_/¯